services/business/checkers/virtual_service_checker.go

```diff
@@ -30,11 +32,24 @@ func (in VirtualServiceChecker) Check() models.IstioValidations {
 // Runs individual checks for each virtual service
 func (in VirtualServiceChecker) runIndividualChecks() models.IstioValidations {
 	validations := models.IstioValidations{}
+	validationsChan := make(chan models.IstioValidations)
+
+	var wg sync.WaitGroup
```

Comment:
I guess this comes from the no_service_checker.go design where both were used to nest concurrent results.

Reply:
Yes but the same can be done with just channels, basically consolidation happens synchronously for every item received through the channel to avoid concurrent writes (see https://github.com/kiali/kiali/pull/334/files )

It performs even a bit better as the merge operation can still run concurrently with the individual checker operations, still without any risk of concurrent writes.